controller/src/test/java/org/jboss/as/controller/persistence/PersistentResourceXMLParserTestCase.java

```diff
@@ -26,73 +26,131 @@
 import org.jboss.as.controller.PersistentResourceDefinition;
 import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.PersistentResourceXMLParser;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.StringListAttributeDefinition;
 import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
 import org.jboss.as.controller.descriptions.NonResolvingResourceDescriptionResolver;
+import org.jboss.as.controller.operations.validation.StringLengthValidator;
 import org.jboss.dmr.ModelNode;
 import org.jboss.dmr.ModelType;
-import org.jboss.staxmapper.XMLElementReader;
-import org.jboss.staxmapper.XMLElementWriter;
-import org.jboss.staxmapper.XMLExtendedStreamReader;
 import org.jboss.staxmapper.XMLExtendedStreamWriter;
 import org.jboss.staxmapper.XMLMapper;
+import org.junit.Assert;
 import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.bootstrap.DOMImplementationRegistry;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSInput;
+import org.w3c.dom.ls.LSParser;
+import org.w3c.dom.ls.LSSerializer;
 
 /**
  * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2015 Red Hat inc.
+ * @author Tomaz Cerar
  */
 public class PersistentResourceXMLParserTestCase {
 
-    @Test
-    public void testParse() throws XMLStreamException {
 
-        MyParser parser = new MyParser();
+    static String readResource(final String name) throws IOException {
 
-        String xml = "<subsystem xmlns=\"" + parser.NAMESPACE + "\">\n" +
-                "  <resource name=\"foo\">\n" +
-                "    <cluster attr1='bar'\n" +
-                "             attr2='baz' />\n" +
-                "    <security my-attr1='alice'\n" +
-                "             my-attr2='bob' />\n" +
-                "   </resource>\n" +
-                "  <resource name=\"foo2\">\n" +
-                "    <cluster attr1='bar2'\n" +
-                "             attr2='baz2' />\n" +
-                "   </resource>\n" +
-                "</subsystem>";
-        System.out.println(xml);
+        URL configURL = PersistentResourceXMLParserTestCase.class.getResource(name);
+        Assert.assertNotNull(name + " url is null", configURL);
+
+        StringWriter writer = new StringWriter();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(configURL.openStream()))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                writer.write(line + "\n");
+            }
+        }
+        return writer.toString();
+    }
+
+    public static XMLExtendedStreamWriter createXMLStreamWriter(XMLStreamWriter writer) throws Exception {
+        // Use reflection to access package protected class FormattingXMLStreamWriter
+        // TODO: at some point the staxmapper API could be enhanced to make this unnecessary
```

Comment:
This shouldn't be necessary any more. But it's fine to clean this up in a separate PR, as there's other code like this.